pkg/planner/core/indexmerge_unfinished_path.go

```diff
@@ -368,39 +366,41 @@ func buildIntoAccessPath(
 				if path == nil {
 					continue
 				}
-				paths = []*util.AccessPath{path}
+				oneAlternative = []*util.AccessPath{path}
 			}
 			needSelection = needSelection || unfinishedPath.needKeepFilter
 
-			alternativesForORBranch = append(alternativesForORBranch, alternative{paths, needSelection})
+			if needSelection {
+				// only need to set one of the paths to true
+				oneAlternative[0].KeepIndexMergeORSourceFilter = true
+			}
+			alternativesForORBranch = append(alternativesForORBranch, oneAlternative)
 		}
 		if len(alternativesForORBranch) == 0 {
 			return nil
 		}
+		allAlternativePaths = append(allAlternativePaths, alternativesForORBranch)
+	}
 
-		bestAlternative := slices.MinFunc(
-			alternativesForORBranch, func(a, b alternative) int {
-				return cmpAlternativesByRowCount(a.paths, b.paths)
-			})
-		if len(bestAlternative.paths) == 0 {
-			return nil
-		}
-		// Succeeded to get valid path(s) for this partial path.
-		partialPaths = append(partialPaths, bestAlternative.paths...)
-		needSelectionGlobal = needSelectionGlobal || bestAlternative.needSelection
+	// 2. TODO
+
+	// 3. Build the final access path.
+	possiblePath := &util.AccessPath{
+		PartialAlternativeIndexPaths: allAlternativePaths,
+		TableFilters:                 slices.Delete(slices.Clone(allConds), orListIdxInAllConds, orListIdxInAllConds+1),
+		IndexMergeORSourceFilter:     allConds[orListIdxInAllConds],
 	}
 
-	// 2. Collect the final table filter
-	// We always put all filters in the top level AND list except for the OR list into the final table filters.
-	// Whether to put the OR list into the table filters also depends on the needSelectionGlobal.
-	tableFilter := slices.Clone(allConds)
-	if !needSelectionGlobal {
-		tableFilter = slices.Delete(tableFilter, orListIdxInAllConds, orListIdxInAllConds+1)
+	// For estimation, we need the decided partial paths. So we use a simple heuristic to choose the partial paths by
+	// comparing the row count just for estimation here.
+	pathsForEstimate := make([]*util.AccessPath, 0, len(allAlternativePaths))
+	for _, oneORBranch := range allAlternativePaths {
+		pathsWithMinRowCount := slices.MinFunc(oneORBranch, cmpAlternativesByRowCount)
+		pathsForEstimate = append(pathsForEstimate, pathsWithMinRowCount...)
 	}
+	possiblePath.CountAfterAccess = estimateCountAfterAccessForIndexMergeOR(ds, pathsForEstimate)
```

Comment:
from the theoretical, we don't need to compute the count after acess at this point cuz it's still undetermined. but its seems to be used to update ds‘s stats introduced by @jiangtao
![image](https://github.com/user-attachments/assets/a1b1bc9e-2a40-4494-ad8c-10f8917eafcb)
by now, it's fine, but not that elegant

Reply:
Yes. Besides this, the calculation of `DataSource.AccessPathMinSelectivity` also needs this field to be correctly set.

Reply:
make sense